src/electron/process_manager.ts

```diff
@@ -388,62 +413,113 @@ function startTun2socks(onDisconnected: () => void): Promise<void> {
-      TUN2SOCKS_TAP_DEVICE_NETWORK}:${TUN2SOCKS_VIRTUAL_ROUTER_NETMASK}`);
-    args.push('--netif-ipaddr', TUN2SOCKS_VIRTUAL_ROUTER_IP);
-    args.push('--netif-netmask', TUN2SOCKS_VIRTUAL_ROUTER_NETMASK);
-    args.push('--socks-server-addr', `${PROXY_IP}:${SS_LOCAL_PORT}`);
-    args.push('--socks5-udp');
-    args.push('--udp-relay-addr', `${PROXY_IP}:${SS_LOCAL_PORT}`);
-    args.push('--loglevel', 'error');
-
-    // TODO: Duplicate ss-local's error handling.
-    try {
-      tun2socks = spawn(pathToEmbeddedExe('badvpn-tun2socks'), args);
-      tun2socks.on('exit', (code, signal) => {
-        if (signal) {
-          // tun2socks exits with SIGTERM when we stop it.
-          console.info(`tun2socks exited with signal ${signal}`);
-        } else {
-          console.info(`tun2socks exited with code ${code}`);
-          if (code === 1) {
-            // tun2socks exits with code 1 upon failure. When the machine sleeps, tun2socks exits
-            // due to a failure to read the tap device.
-            // Restart tun2socks with a timeout so the event kicks in when the device wakes up.
-            console.info('Restarting tun2socks...');
-            setTimeout(() => {
-              startTun2socks(onDisconnected)
-                .then(() => {
-                  resolve();
-                })
-                .catch((e) => {
-                  console.error('Failed to restart tun2socks');
-                  onDisconnected();
-                  teardownVpn();
-                });
-            }, 3000);
-            return;
+    if (tunDeviceName !== undefined) {
+      const args: string[] = [];
+      args.push('--tundev', tunDeviceName);
+      args.push('--netif-ipaddr', TUN2SOCKS_VIRTUAL_ROUTER_IP);
+      args.push('--netif-netmask', TUN2SOCKS_VIRTUAL_ROUTER_NETMASK);
+      args.push('--socks-server-addr', `${PROXY_IP}:${SS_LOCAL_PORT}`);
+      args.push('--socks5-udp');
+      args.push('--udp-relay-addr', `${PROXY_IP}:${SS_LOCAL_PORT}`);
+      args.push('--loglevel', 'error');
+
+      let tun2socksFilename = 'badvpn-tun2socks';
+      if (isWindows) {
+        tun2socksFilename = pathToEmbeddedExe(tun2socksFilename);
+      } else if (isLinux) {
+        tun2socksFilename = pathToEmbeddedBinary(tun2socksFilename);
+      }
+      console.log(tun2socksFilename);
+
+      // TODO: Duplicate ss-local's error handling.
+      try {
+        tun2socks = spawn(tun2socksFilename, args);
+
+        // Ignore stdio if not consuming the process output (pass {stdio: 'igonore'} to spawn);
+        // otherwise the process execution is suspended when the unconsumed streams exceed the
+        // system limit (~200KB). See https://github.com/nodejs/node/issues/4236
+        //
+        // AZ - In *nix world std*'s are buffered so we can't use this method
+        // added a delay after this for now. We need a better solution - maybe node-ps
+        tun2socks.stdout.on('data', (s) => {
```

Comment:
The Windows client doesn't wait for stdout from tun2socks - I think this comment would be better placed on `startShadowsocks`.